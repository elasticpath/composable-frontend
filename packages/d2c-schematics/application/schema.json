{
  "$schema": "http://json-schema.org/draft-07/schema",
  "$id": "SchematicsAngularApp",
  "title": "Angular Application Options Schema",
  "type": "object",
  "description": "Generates a new basic application definition in the \"projects\" subfolder of the workspace.",
  "additionalProperties": false,
  "properties": {
    "projectRoot": {
      "description": "The root directory of the new app.",
      "type": "string",
      "visible": false
    },
    "name": {
      "description": "The name of the new app.",
      "type": "string",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "What name would you like to use for the application?"
    },
    "skipTests": {
      "description": "Do not create \"spec.ts\" test files for the application.",
      "type": "boolean",
      "default": false,
      "alias": "S"
    }
  },
  "required": ["name"]
}
