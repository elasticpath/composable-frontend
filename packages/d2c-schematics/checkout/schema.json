{
  "$schema": "http://json-schema.org/draft-07/schema",
  "$id": "SchematicsCheckout",
  "title": "Checkout Options Schema",
  "type": "object",
  "description": "Generates checkout page and supporting functionality in the workspace.",
  "additionalProperties": true,
  "properties": {
    "skipTests": {
      "description": "Do not create \"spec.ts\" test files for the application.",
      "type": "boolean",
      "default": false,
      "alias": "S"
    },
    "skipConfig": {
      "description": "Do not configure Checkout in Elastic Path Commerce Cloud.",
      "type": "boolean",
      "default": false
    },
    "directory": {
      "type": "string",
      "description": "The directory name to create the workspace in."
    },
    "path": {
      "type": "string",
      "format": "path",
      "$default": {
        "$source": "workingDirectory"
      },
      "description": "The path at which to create the component file, relative to the current workspace. Default is a folder with the same name as the component in the project root.",
      "visible": false
    },
    "paymentGatewayType": {
      "description": "The type of payment gateway to use for your checkout experience.",
      "type": "string",
      "enum": ["EP Payments", "None"],
      "x-prompt": "Which payment gateway do you want to use?"
    },
    "epccClientId": {
      "description": "The client id value for an Elastic Path Commerce Cloud store.",
      "type": "string",
      "visible": "false"
    },
    "epccClientSecret": {
      "description": "The client secret value for an Elastic Path Commerce Cloud store.",
      "type": "string",
      "visible": "false"
    },
    "epccEndpointUrl": {
      "description": "The geographical endpoint url for an Elastic Path Commerce Cloud store.",
      "type": "string",
      "visible": "false"
    }
  },
  "required": ["epccClientId", "epccClientSecret", "epccEndpointUrl", "directory"]
}
