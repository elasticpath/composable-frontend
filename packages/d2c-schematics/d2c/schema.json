{
  "$schema": "http://json-schema.org/draft-07/schema",
  "$id": "SchematicsD2C",
  "title": "Elastic Path Commerce Cloud Direct to Consumer (D2C) Options Schema",
  "type": "object",
  "description": "Creates a new Elastic Path Commerce Cloud storefront by combining the workspace and application schematics.",
  "additionalProperties": true,
  "properties": {
    "directory": {
      "type": "string",
      "description": "The directory name to create the workspace in."
    },
    "name": {
      "description": "The name of the new workspace and project.",
      "type": "string",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "What name would you like to use for the new workspace and initial project?"
    },
    "packageManager": {
      "description": "Node package manager to use.",
      "type": "string",
      "enum": ["npm", "yarn", "pnpm"],
      "default": "npm"
    },
    "epccClientId": {
      "description": "The client id value for an Elastic Path Commerce Cloud store.",
      "type": "string",
      "x-prompt": "What is the client id for your store?"
    },
    "epccClientSecret": {
      "description": "The client secret value for an Elastic Path Commerce Cloud store.",
      "type": "string",
      "x-prompt": "What is the client secret for your store?"
    },
    "epccEndpointUrl": {
      "description": "The geographical endpoint url for an Elastic Path Commerce Cloud store.",
      "type": "string",
      "default": "api.moltin.com",
      "x-prompt": "What is the endpoint for your store?"
    },
    "skipInstall": {
      "description": "Do not install dependency packages.",
      "type": "boolean",
      "default": false
    },
    "skipGit": {
      "description": "Do not initialize a git repository.",
      "type": "boolean",
      "default": false,
      "alias": "g"
    },
    "commit": {
      "description": "Initial git repository commit information.",
      "oneOf": [
        { "type": "boolean" },
        {
          "type": "object",
          "properties": {
            "name": {
              "type": "string"
            },
            "email": {
              "type": "string",
              "format": "email"
            },
            "message": {
              "type": "string"
            }
          },
          "required": ["name", "email"]
        }
      ],
      "default": true
    },
    "skipTests": {
      "description": "Do not generate \"spec.ts\" test files for the new project.",
      "type": "boolean",
      "default": false,
      "alias": "S"
    },
    "plpType": {
      "description": "The type of product list page to use.",
      "type": "string",
      "enum": ["Algolia", "None"]
    }
  },
  "required": ["name", "epccClientId", "epccClientSecret", "epccEndpointUrl"]
}
