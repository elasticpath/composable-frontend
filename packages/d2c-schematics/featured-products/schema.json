{
  "$schema": "http://json-schema.org/draft-07/schema",
  "$id": "FeaturedProductsSchema",
  "title": "Featured Products Schematics Schema",
  "type": "object",
  "additionalProperties": false,
  "properties": {
    "path": {
      "type": "string",
      "format": "path",
      "$default": {
        "$source": "workingDirectory"
      },
      "description": "The path at which to create the component file, relative to the current workspace. Default is a folder with the same name as the component in the project root.",
      "visible": false
    },
    "featured-node-id": {
      "description": "The id of the EPCC node entity you want to use to power the featured product component",
      "type": "string",
      "x-prompt": "What is the id of the node you want to use for featured products?"
    }
  },
  "required": ["featured-node-id"]
}
