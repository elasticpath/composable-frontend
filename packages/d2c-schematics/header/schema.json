{
  "$schema": "http://json-schema.org/draft-07/schema",
  "$id": "HeaderSchema",
  "title": "Header Schematics Schema",
  "type": "object",
  "additionalProperties": false,
  "properties": {
    "path": {
      "type": "string",
      "format": "path",
      "$default": {
        "$source": "workingDirectory"
      },
      "description": "The path at which to create the component file, relative to the current workspace. Default is a folder with the same name as the component in the project root.",
      "visible": false
    },
    "search": {
      "description": "Should include the search functionality",
      "type": "boolean",
      "default": true
    }
  }
}
