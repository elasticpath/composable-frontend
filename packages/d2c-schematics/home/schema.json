{
  "$schema": "http://json-schema.org/draft-07/schema",
  "$id": "HomeSchema",
  "title": "Home Schematics Schema",
  "type": "object",
  "additionalProperties": false,
  "properties": {
    "path": {
      "type": "string",
      "format": "path",
      "$default": {
        "$source": "workingDirectory"
      },
      "description": "The path at which to create the component file, relative to the current workspace. Default is a folder with the same name as the component in the project root.",
      "visible": false
    },
    "components": {
      "description": "The home page components to include.",
      "type": "array",
      "uniqueItems": true,
      "items": {
        "enum": ["PromotionBanner", "FeaturedProducts"],
        "type": "string"
      },
      "default": [],
      "x-prompt": "What home page components would you like to include?"
    }
  }
}
