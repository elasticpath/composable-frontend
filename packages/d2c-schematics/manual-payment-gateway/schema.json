{
  "$schema": "http://json-schema.org/draft-07/schema",
  "$id": "ManualPaymentGatewayCheckout",
  "title": "Manual Payment Gateway Options Schema",
  "type": "object",
  "description": "Generates checkout page and supporting functionality in the workspace.",
  "additionalProperties": true,
  "properties": {
    "skipTests": {
      "description": "Do not create \"spec.ts\" test files for the application.",
      "type": "boolean",
      "default": false,
      "alias": "S"
    },
    "skipConfig": {
      "description": "Do not configure Checkout in Elastic Path Commerce Cloud.",
      "type": "boolean",
      "default": false
    },
    "directory": {
      "type": "string",
      "description": "The directory name to create the workspace in."
    },
    "path": {
      "type": "string",
      "format": "path",
      "$default": {
        "$source": "workingDirectory"
      },
      "description": "The path at which to create the component file, relative to the current workspace. Default is a folder with the same name as the component in the project root.",
      "visible": false
    },
    "epccClientId": {
      "description": "The client id value for an Elastic Path Commerce Cloud store.",
      "type": "string",
      "visible": "false"
    },
    "epccClientSecret": {
      "description": "The client secret value for an Elastic Path Commerce Cloud store.",
      "type": "string",
      "visible": "false"
    },
    "epccEndpointUrl": {
      "description": "The geographical endpoint url for an Elastic Path Commerce Cloud store.",
      "type": "string",
      "visible": "false"
    }
  },
  "required": ["epccClientId", "epccClientSecret", "epccEndpointUrl", "directory"]
}
