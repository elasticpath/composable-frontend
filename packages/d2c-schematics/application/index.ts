import { normalize } from "@angular-devkit/core"
import {
  MergeStrategy,
  Rule,
  Tree,
  apply,
  applyTemplates,
  chain,
  mergeWith,
  move,
  strings,
  url,
} from "@angular-devkit/schematics"
import { Schema as ApplicationOptions } from "./schema"

export default function (options: ApplicationOptions): Rule {
  return async (_host: Tree) => {
    const appDir = normalize(options.projectRoot || "")

    return chain([
      mergeWith(
        apply(url("./files"), [
          applyTemplates({
            utils: strings,
            ...options,
          }),
          move(appDir),
        ]),
        MergeStrategy.Overwrite
      ),
    ])
  }
}
