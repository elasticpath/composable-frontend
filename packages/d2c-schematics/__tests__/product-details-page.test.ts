import {
  SchematicTestRunner,
  UnitTestTree,
} from "@angular-devkit/schematics/testing"

import { Schema as WorkspaceOptions } from "../workspace/schema"
import { Schema as ApplicationOptions } from "../application/schema"

describe("Product Details Page Schematic", () => {
  const schematicRunner = new SchematicTestRunner(
    "@schematics/angular",
    require.resolve("../collection.json"),
  )

  const workspaceOptions: WorkspaceOptions = {
    name: "workspace",
    epccClientId: "123",
    epccClientSecret: "456",
    epccEndpointUrl: "euwest.api.elasticpath.com",
  }

  const applicationOptions: ApplicationOptions = {
    name: "foo",
  }

  const defaultOptions = {}

  let initTree: UnitTestTree
  beforeEach(async () => {
    /**
     * Product Details Page schematic depends on workspace and application schematics
     */
    const workspaceTree = await schematicRunner.runSchematic(
      "workspace",
      workspaceOptions,
    )

    initTree = await schematicRunner.runSchematic(
      "application",
      applicationOptions,
      workspaceTree,
    )
  })

  it("should create product detail component files", async () => {
    const options = { ...defaultOptions }
    const tree = await schematicRunner.runSchematic(
      "product-details-page",
      options,
      initTree,
    )

    const files = tree.files
    // console.log("files: ", JSON.stringify(files))
    expect(files).toMatchSnapshot()
  })

  it("should have files provided by application schematic", async () => {
    const options = { ...defaultOptions }
    const tree = await schematicRunner.runSchematic(
      "product-details-page",
      options,
      initTree,
    )

    const files = tree.files

    expect(files).toIncludeAllPartialMembers([
      "/src/components/shimmer.tsx",
      "/src/lib/epcc-implicit-client.ts",
      "/src/lib/to-base-64.ts",
      "/src/lib/custom-rule-headers.ts",
      "/src/lib/is-empty-object.ts",
    ])
  })
})
