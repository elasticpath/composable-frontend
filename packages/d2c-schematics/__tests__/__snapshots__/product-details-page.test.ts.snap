// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product Details Page Schematic should create product detail component files 1`] = `
[
  "/README.md",
  "/.composablerc",
  "/.env.example",
  "/.env.local",
  "/.env.test",
  "/.eslintrc.json",
  "/.gitignore",
  "/.lintstagedrc.js",
  "/.prettierignore",
  "/.prettierrc",
  "/license.md",
  "/next-env.d.ts",
  "/next.config.js",
  "/package.json",
  "/playwright.config.ts",
  "/postcss.config.js",
  "/tailwind.config.ts",
  "/tsconfig.json",
  "/vite.config.ts",
  "/public/favicon.ico",
  "/e2e/product-details-page.spec.ts",
  "/e2e/util/epcc-admin-client.ts",
  "/e2e/util/epcc-client.ts",
  "/e2e/util/fill-form-field.ts",
  "/e2e/util/has-published-catalog.ts",
  "/e2e/util/missing-published-catalog.ts",
  "/e2e/util/skip-ci-env.ts",
  "/e2e/util/resolver-product-from-store.ts",
  "/e2e/models/d2c-product-detail-page.ts",
  "/src/middleware.ts",
  "/src/app/error.tsx",
  "/src/app/layout.tsx",
  "/src/app/not-found.tsx",
  "/src/app/providers.tsx",
  "/src/app/(store)/layout.tsx",
  "/src/app/(store)/not-found.tsx",
  "/src/app/(store)/about/page.tsx",
  "/src/app/(store)/faq/page.tsx",
  "/src/app/(store)/shipping/page.tsx",
  "/src/app/(store)/terms/page.tsx",
  "/src/app/(store)/products/[productId]/page.tsx",
  "/src/app/(store)/products/[productId]/product-display.tsx",
  "/src/app/configuration-error/page.tsx",
  "/src/components/Checkbox.tsx",
  "/src/components/LoadingDots.tsx",
  "/src/components/Spinner.tsx",
  "/src/components/shimmer.tsx",
  "/src/components/NoImage.tsx",
  "/src/components/accordion/Accordion.tsx",
  "/src/components/alert/Alert.tsx",
  "/src/components/button/Button.tsx",
  "/src/components/button/FormStatusButton.tsx",
  "/src/components/button/LoaderIcon.tsx",
  "/src/components/button/StatusButton.tsx",
  "/src/components/button/TextButton.tsx",
  "/src/components/form/Form.tsx",
  "/src/components/icons/ep-icon.tsx",
  "/src/components/icons/ep-logo.tsx",
  "/src/components/icons/github-icon.tsx",
  "/src/components/input/Input.tsx",
  "/src/components/label/Label.tsx",
  "/src/components/navigation-menu/NavigationMenu.tsx",
  "/src/components/number-input/EditQuantityButton.tsx",
  "/src/components/number-input/NumberInput.tsx",
  "/src/components/number-input/use-counter.ts",
  "/src/components/pagination/Pagination.tsx",
  "/src/components/pagination/ResourcePagination.tsx",
  "/src/components/radio-group/RadioGroup.tsx",
  "/src/components/select/Select.tsx",
  "/src/components/separator/Separator.tsx",
  "/src/components/shared/blurb.tsx",
  "/src/components/sheet/Sheet.tsx",
  "/src/components/skeleton/Skeleton.tsx",
  "/src/components/toast/toaster.tsx",
  "/src/components/product/Price.tsx",
  "/src/components/product/ProductDetails.tsx",
  "/src/components/product/ProductExtensions.tsx",
  "/src/components/product/ProductSummary.tsx",
  "/src/components/product/SimpleProduct.tsx",
  "/src/components/product/StrikePrice.tsx",
  "/src/components/product/bundles/BundleProduct.tsx",
  "/src/components/product/bundles/ProductComponent.tsx",
  "/src/components/product/bundles/ProductComponents.tsx",
  "/src/components/product/bundles/form-parsers.test.ts",
  "/src/components/product/bundles/form-parsers.ts",
  "/src/components/product/bundles/sort-by-order.ts",
  "/src/components/product/bundles/validation-schema.test.ts",
  "/src/components/product/bundles/validation-schema.ts",
  "/src/components/product/carousel/CarouselListener.tsx",
  "/src/components/product/carousel/HorizontalCarousel.tsx",
  "/src/components/product/carousel/ProductCarousel.module.css",
  "/src/components/product/carousel/ProductCarousel.tsx",
  "/src/components/product/carousel/ProductHighlightCarousel.tsx",
  "/src/components/product/variations/ProductVariationColor.tsx",
  "/src/components/product/variations/ProductVariationStandard.tsx",
  "/src/components/product/variations/ProductVariations.tsx",
  "/src/components/product/variations/VariationProduct.tsx",
  "/src/lib/build-site-navigation.ts",
  "/src/lib/cn.tsx",
  "/src/lib/constants.ts",
  "/src/lib/cookie-constants.ts",
  "/src/lib/custom-rule-headers.ts",
  "/src/lib/epcc-errors.ts",
  "/src/lib/epcc-implicit-client.ts",
  "/src/lib/epcc-server-client.ts",
  "/src/lib/epcc-server-side-credentials-client.ts",
  "/src/lib/epcc-server-side-implicit-client.ts",
  "/src/lib/form-url-encode-body.ts",
  "/src/lib/format-currency.tsx",
  "/src/lib/format-iso-date-string.ts",
  "/src/lib/get-error-message.ts",
  "/src/lib/get-store-initial-state.ts",
  "/src/lib/is-account-member-authenticated.ts",
  "/src/lib/is-empty-object.ts",
  "/src/lib/resolve-cart-env.ts",
  "/src/lib/resolve-ep-currency-code.ts",
  "/src/lib/resolve-epcc-env.ts",
  "/src/lib/retrieve-account-member-credentials.ts",
  "/src/lib/sort-alphabetically.ts",
  "/src/lib/to-base-64.ts",
  "/src/lib/token-expired.ts",
  "/src/lib/color-lookup.ts",
  "/src/lib/connect-products-with-main-images.ts",
  "/src/lib/is-supported-extension.ts",
  "/src/lib/product-context.ts",
  "/src/lib/product-helper.test.ts",
  "/src/lib/product-helper.ts",
  "/src/lib/product-util.test.ts",
  "/src/lib/product-util.ts",
  "/src/lib/middleware/apply-set-cookie.ts",
  "/src/lib/middleware/cart-cookie-middleware.ts",
  "/src/lib/middleware/create-missing-environment-variable-url.ts",
  "/src/lib/middleware/implicit-auth-middleware.ts",
  "/src/lib/middleware/middleware-runner.ts",
  "/src/lib/types/deep-partial.ts",
  "/src/lib/types/non-empty-array.ts",
  "/src/lib/types/read-only-non-empty-array.ts",
  "/src/lib/types/matrix-object-entry.ts",
  "/src/lib/types/product-types.ts",
  "/src/services/hierarchy.ts",
  "/src/services/products.ts",
  "/src/styles/globals.css",
]
`;
